test/ValidateOAuthVersionMessageHandler.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace OAuth.Net.Tests
+{
+    public class ValidateOAuthVersionMessageHandler
+    {
+        [Fact]
+        public async Task ValidateDefaultOAuthVersionHandler()
+        {
+            using (var handler = new OAuthMessageHandler("apiKey", "secret", "authToken", "authTokenSecret"))
+            {
+                var oauthVersion = await GetOAuthVersionFromHandlerAsync(handler);
+
+                Assert.Equal("1.0", oauthVersion);
+            }
+        }
+
+        [Fact]
+        public async Task ValidateOAuthVersionOmitHandler()
+        {
+            using (var handler = new OAuthMessageHandler("apiKey", "secret", "authToken", "authTokenSecret", OAuthVersion.Omit))
+            {
+                var oauthVersion = await GetOAuthVersionFromHandlerAsync(handler);
+
+                Assert.Null(oauthVersion);
+            }
+        }
+
+        [Fact]
+        public async Task ValidateOAuthVersionOneZeroHandler()
+        {
+            using (var handler = new OAuthMessageHandler("apiKey", "secret", "authToken", "authTokenSecret", OAuthVersion.OneZero))
+            {
+                var oauthVersion = await GetOAuthVersionFromHandlerAsync(handler);
+
+                Assert.Equal("1.0", oauthVersion);
+            }
+        }
+
+        [Fact]
+        public async Task ValidateOAuthVersionOneZeroAHandler()
+        {
+            using (var handler = new OAuthMessageHandler("apiKey", "secret", "authToken", "authTokenSecret", OAuthVersion.OneZeroA))
+            {
+                var oauthVersion = await GetOAuthVersionFromHandlerAsync(handler);
+
+                Assert.Equal("1.0a", oauthVersion);
+            }
+        }
+
+        private async Task<string> GetOAuthVersionFromHandlerAsync(OAuthMessageHandler handler)
```

Comment:
Is this trying to add a test that will be more of an end-to-end scenario? Otherwise you can simply compute the oauth signature and extract the version information from there.

If this is an End-To-End scenario we should add a trait to the tests (so there is a way to just run tests that don't hit the network).

Reply:
It is intended to be more of a unit test, but I personally prefer not call private methods (through reflection) in my tests, so therefore it became more of an end-to-end scenario (I'm not even calling the tested method directly, I let the `HttpClient` do it for me). As stated in the comment above this test should not hit the network though, due to the `TestHttpMessageHandler`.

Reply:
I agree with you :) -- using reflection to call private methods for testing is not the right approach.

I think an option for testing the handler would be to subclass the OAuthMessageHandler and intercept the calls to actually hit the network (similar to what you did here).

I'll file an issue to track that test change.